src/Orc.CsvHelper/Orc.CsvHelper.Shared/Services/CsvWriterService.cs

```diff
@@ -124,6 +144,53 @@ public virtual void WriteCsv<TRecord>(IEnumerable<TRecord> records, string csvFi
             WriteCsv(records, csvFilePath, typeof(TRecord), csvMap, csvConfiguration, throwOnError, cultureInfo);
         }
 
+        public virtual async Task WriteCsvAsync<TRecord>(IEnumerable<TRecord> records, string csvFilePath, CsvClassMap csvMap, CsvConfiguration csvConfiguration = null, bool throwOnError = false, CultureInfo cultureInfo = null)
+        {
+            await WriteCsvAsync(records, csvFilePath, typeof(TRecord), csvMap, csvConfiguration, throwOnError, cultureInfo);
+        }
+
+        public virtual async Task WriteCsvAsync(IEnumerable records, string csvFilePath, Type recordType, CsvClassMap csvMap, CsvConfiguration csvConfiguration = null, bool throwOnError = false, CultureInfo cultureInfo = null)
+        {
+            byte[] array = null;
+            await TaskHelper.RunAndWaitAsync(() =>
+            {
+                using (var memoryStream = new MemoryStream())
+                {
+                    WriteCsv(records, new StreamWriter(memoryStream), recordType, csvMap, csvConfiguration, throwOnError, cultureInfo);
+                    array = memoryStream.ToArray();
+                }
+            });
+
+            using (var fileStream = _fileService.Create(csvFilePath))
+            {
+                await fileStream.WriteAsync(array, 0, array.Length);
+            }
+        }
+
+        public virtual async Task WriteCsvAsync<TRecord, TMap>(IEnumerable<TRecord> records, string csvFilePath, CsvConfiguration csvConfiguration = null, bool throwOnError = false, CultureInfo cultureInfo = null)
+            where TMap : CsvClassMap
+        {
+            await WriteCsvAsync(records, csvFilePath, typeof(TMap), csvConfiguration, throwOnError, cultureInfo);
+        }
+
+        public virtual async Task WriteCsvAsync<TRecord>(IEnumerable<TRecord> records, string csvFilePath, Type csvMap = null, CsvConfiguration csvConfiguration = null, bool throwOnError = false, CultureInfo cultureInfo = null)
+        {
+            byte[] array = null;
+            await TaskHelper.RunAndWaitAsync(() =>
```

Comment:
yes... i thought about that (it just swapping the thread).. but it wouldn't block UI though... i hesitated at this point)... ok, will remove this one